packages/@uppy/companion/src/server/provider/webdav/common.js

```diff
@@ -0,0 +1,125 @@
+const Provider = require('../Provider')
+const logger = require('../../logger')
+const { getProtectedHttpAgent, validateURL } = require('../../helpers/request')
+const { ProviderApiError, ProviderAuthError } = require('../error')
+
+/**
+ * WebdavProvider base class provides implementations that could be shared by simple and oauth providers
+ */
+class WebdavProvider extends Provider {
+  async getClientHelper ({ url, ...options }) {
+    const { allowLocalUrls } = this
+    if (!validateURL(url, allowLocalUrls)) {
+      throw new Error('invalid webdav url')
+    }
+    const { protocol } = new URL(url)
+    const HttpAgentClass = getProtectedHttpAgent({ protocol, allowLocalIPs: !allowLocalUrls })
+
+    const { createClient } = await import('webdav')
```

Comment:
Any reason why the import is done in this function call instead of global?

Reply:
tbh I don't remember at all. Maybe I copied it from some other provider, maybe I had issues with typescript types, ... will try to make it a regular import

Reply:
AH, yes! The webdav package is ESM and companion uses CommonJS - hence dynamic async import needs to be used (or Node 20.17 or 22)

Maybe I should add a comment for that

Reply:
Node.js LTS now has [stable require(esm)](https://github.com/nodejs/node/pull/55217) so in the future when we drop versions we can just import it :)